api/py/ai/chronon/repo/validator.py

```diff
@@ -221,6 +309,27 @@ def _validate_group_by(self, group_by: GroupBy) -> List[str]:
             else:
                 group_by.metaData.production = True
 
+        # validate the derivations are defined correctly
+        if group_by.derivations:
+            columns = set(get_pre_derived_group_by_columns(group_by))
+            derived_columns = set()
+            for derivation in group_by.derivations:
+                if derivation.name in derived_columns:
+                    errors.append(
+                        "Incorrect derivation name {} due to output column name conflict".format(derivation.name))
+                else:
+                    derived_columns.add(derivation.name)
+            for derivation in group_by.derivations:
+                dev_name = derivation.name
```

Comment:
```suggestion
                der_name = derivation.name
```

Reply:
dev has another meaning

Reply:
actually, let's use full name: derived_name to avoid confusion

Reply:
updated